core/client-internal/src/main/java/alluxio/client/UfsUtils.java

```diff
@@ -42,177 +40,70 @@
-   *        system
-   * @param configuration the instance of {@link Configuration} to be used
-   * @throws IOException when an event that prevents the operation from completing is encountered
-   * @throws AlluxioException if an unexpected Alluxio error occurs
-   */
-  private static void loadUfs(AlluxioURI tfsAddrRootPath, AlluxioURI ufsAddrRootPath,
-      String excludePaths, Configuration configuration) throws IOException, AlluxioException {
-    FileSystem tfs = FileSystem.Factory.get();
-
-    PrefixList excludePathPrefix = new PrefixList(excludePaths, ";");
-
-    loadUfs(tfs, tfsAddrRootPath, ufsAddrRootPath, excludePathPrefix, configuration);
-  }
-
-  /**
-   * Loads files under path "ufsAddress/ufsRootPath" (excluding excludePathPrefix) to the given fs
-   * under the given tfsRootPath directory.
-   *
-   * @param fs the {@link FileSystem} handler created out of address like "alluxio://host:port"
-   * @param alluxioPath the destination point in Alluxio file system to load the under file system
-   *        path onto
-   * @param ufsAddrRootPath the address and root path of the under file system, like
-   *        "hdfs://host:port/dir"
-   * @param excludePathPrefix paths to exclude from ufsRootPath, which will not be registered in
-   *        Alluxio file system
+   * @param mountPointUri the destination point in Alluxio file system to load the under file system
+   *        path onto, this must not already exist
+   * @param mountDirectoryUri the directory in the under file system to mount
+   * @param exclusions prefixes to exclude which will not be registered in the Alluxio file system
    * @param conf instance of {@link Configuration}
-   * @throws IOException when an event that prevents the operation from completing is encountered
-   * @throws AlluxioException if an unexpected alluxio error occurs
-   * @deprecated As of version 0.8. Use
-   *             {@link #loadUfs(AlluxioURI, AlluxioURI, String, Configuration)} instead.
+   * @throws IOException if an error occurs when operating on the under file system
+   * @throws AlluxioException if an unexpected Alluxio error occurs
    */
-  @Deprecated
-  public static void loadUfs(FileSystem fs, AlluxioURI alluxioPath, AlluxioURI ufsAddrRootPath,
-      PrefixList excludePathPrefix, Configuration conf) throws IOException, AlluxioException {
-    LOG.info("Loading to {} {} {}", alluxioPath, ufsAddrRootPath, excludePathPrefix);
-    try {
-      // resolve and replace hostname embedded in the given ufsAddress/alluxioAddress
-      ufsAddrRootPath = NetworkAddressUtils.replaceHostName(ufsAddrRootPath);
-      alluxioPath = NetworkAddressUtils.replaceHostName(alluxioPath);
-    } catch (UnknownHostException e) {
-      LOG.error("Failed to resolve hostname", e);
-      throw new IOException(e);
+  public static void loadUfs(AlluxioURI mountPointUri, AlluxioURI mountDirectoryUri,
```

Comment:
I would still want to mark this as deprecated. loadUfs is a shorthand for "fs mount" followed by "fs loadMetadata -R" ... all we need to get rid of loadUfs is to support -R for the loadMetadata shell command.